packages/ember-glimmer/tests/integration/helpers/element-action-test.js

```diff
@@ -0,0 +1,1340 @@
+
+    let ExampleComponent = Component.extend({
+      actions: {
+        foo() {
+          fooCallCount++;
+        }
+      }
+    });
+
+    this.registerComponent('example-component', {
+      ComponentClass: ExampleComponent,
+      template: '<button {{action "foo"}}>Click me</button>'
+    });
+
+    this.render('{{example-component}}');
+
+    this.assert.equal(fooCallCount, 0, 'foo has not been called');
+
+    this.runTask(() => this.rerender());
+
+    this.assert.equal(fooCallCount, 0, 'foo has not been called');
+
+    this.runTask(() => {
+      this.$('button').click();
+    });
+
+    this.assert.equal(fooCallCount, 1, 'foo has been called 1 time');
+
+    this.runTask(() => {
+      this.$('button').click();
+    });
+
+    this.assert.equal(fooCallCount, 2, 'foo has been called 2 times');
+  }
+
+  ['@test it can call an action with parameters']() {
+    let fooArgs = [];
+    let component;
+
+    let ExampleComponent = Component.extend({
+      member: 'a',
+      init() {
+        this._super(...arguments);
+        component = this;
+      },
+      actions: {
+        foo(thing) {
+          fooArgs.push(thing);
+        }
+      }
+    });
+
+    this.registerComponent('example-component', {
+      ComponentClass: ExampleComponent,
+      template: '<button {{action "foo" member}}>Click me</button>'
+    });
+
+    this.render('{{example-component}}');
+
+    this.assert.deepEqual(fooArgs, [], 'foo has not been called');
+
+    this.runTask(() => this.rerender());
+
+    this.assert.deepEqual(fooArgs, [], 'foo has not been called');
+
+    this.runTask(() => {
+      this.$('button').click();
+    });
+
+    this.assert.deepEqual(fooArgs, ['a'], 'foo has not been called');
+
+    this.runTask(() => {
+      component.set('member', 'b');
+    });
+
+    this.runTask(() => {
+      this.$('button').click();
+    });
+
+    this.assert.deepEqual(fooArgs, ['a', 'b'], 'foo has been called with an updated value');
+  }
+
+  ['@htmlbars it should output a data attribute with a guid']() {
+    this.render('<button {{action "show"}}>me the money</button>');
+
+    let button = this.$('button');
+
+    this.assert.ok(button.attr('data-ember-action').match(/\d+/), 'A data-ember-action attribute with a guid was added');
+  }
+
+  ['@glimmer it should output a marker data attribute and a data attribute with a guid']() {
+    this.render('<button {{action "show"}}>me the money</button>');
+
+    let button = this.$('button');
+    let attributes = getActionAttributes(button.get(0));
+
+    this.assert.ok(button.attr('data-ember-action').match(''), 'An empty data-ember-action attribute was added');
+    this.assert.ok(attributes[0].match(/data-ember-action-\d+/), 'A data-ember-action-xyz attribute with a guid was added');
+  }
+
+  ['@skip it should by default register a click event']() {
+    // TODO
+    // Should we be testing implementation details? Why?
+  }
+
+  ['@skip it should allow alternative events to be handled']() {
```

Comment:
I guess what I'm getting at here is the original test for this intercepts/inspects the slots inside of `ActionManager.registeredActions`. I think a better way to test this would be to set up an action for several events, test each one, and assert that it fires for the respective event.